Adds new Advanced Behavioral Detections section (#4296)
elastic/security-docs: docs/advanced-entity-analytics/advanced-behavioral-detections.asciidoc

[[advanced-behavioral-detections]]

> = Advanced Behavioral Detections

Author: To close the loop on this, per the slack convo, we're lowercasing both `advanced behavioral detections` and `entity risk scoring`.

> Use Elastic's {ml} capabilities and advanced correlation, scoring, and visualization techniques to identify behavioral threats that are most likely to be involved in security incidents in your environment.

Reviewer: ```suggestion
Elastic's {ml} capabilities and advanced correlation, scoring, and visualization techniques can help you identify potential behavioral threats that may be associated with security incidents.
```

> Advanced Behavioral Detections provide two key capabilities:

Reviewer: ```suggestion
Advanced Behavioral Detections includes two key capabilities:
```
